webSites/webSites-war/src/main/java/org/silverpeas/components/websites/servlets/design/SiteDesignActionHandler.java

```diff
@@ -0,0 +1,660 @@
+    String name = request.getParameter("name");
+    // = changement de noms des fichiers et repertoires
+    String newName = request.getParameter("newName");
+
+    // RENAME FOLDER -------------------------------------------------------------
+
+          /* Modifier le nom du repertoire */
+    controller.renameFolder(currentPath + "/" + name, currentPath + "/" + newName);
+
+    request.setAttribute("Site", controller.getSessionSite());
+  }
+
+  private void addFolderInSite(final WebSiteSessionController controller, final HttpRequest request,
+      final String currentPath) throws WebSitesException {
+    // = null la premiere fois, puis = nom du repertoire courant
+    String name = request.getParameter("name");
+
+    // ADD FOLDER -------------------------------------------------------------
+          /* Creer le nouveau repertoire */
+    controller.createFolder(currentPath + "/" + name);
+
+    request.setAttribute("Site", controller.getSessionSite());
+
+  }
+
+  private void updateDesignedSite(final WebSiteSessionController controller, final HttpRequest request,
+      String currentPath) throws WebSitesException {
+    // = rempli au premier acces a designSite pui toujours null
+    String nomSite = request.getParameter(WebSitesRequestRouter.SITE_NAME_PARAM);
+
+    // = rempli la premiere fois a la creation, puis toujours null
+    String description = request.getParameter(WebSitesRequestRouter.DESCRIPTION_PARAM);
+
+    // = rempli la premiere fois a la creation, puis toujours null
+    String nomPage = request.getParameter(WebSitesRequestRouter.PAGE_NAME_PARAM);
+    String tempPopup = request.getParameter(WebSitesRequestRouter.POPUP_PARAM);
+    String etat = request.getParameter("etat");
+    // = rempli la premiere fois a la creation, puis toujours null
+    String listeIcones = request.getParameter(WebSitesRequestRouter.ICON_LIST_PARAM);
+
+    int popup = (tempPopup != null) && (tempPopup.length() > 0) ? 1 : 0;
+    ArrayList<String> listIcons = new ArrayList<>();
+    if (listeIcones != null) {
+      for (String icon : listeIcones.split(",")) {
+        if (!icon.isEmpty()) {
+          listIcons.add(icon);
+        }
+      }
+    }
+
+    SiteDetail ancien = controller.getSessionSite();
+    String id = ancien.getSitePK().getId();
+    int type = ancien.getType();
+
+          /* verif que le nom de la page principale est correcte */
+    boolean searchOk = false;
+    Collection<File> collPages = controller.getAllWebPages2(currentPath);
+    for (File f : collPages) {
+      if (f.getName().equals(nomPage)) {
+        searchOk = true;
+        break;
+      }
+    }
+
+    SiteDetail descriptionSite2 =
+        new SiteDetail(id, controller.getComponentId(), nomSite, description, nomPage, type, null, null,
+            Integer.parseInt(etat), popup);
+
+    if (searchOk) {
+            /* update description en BD */
+      controller.updateWebSite(descriptionSite2);
+
+      if (listIcons.size() > 0) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Use isEmpty() to check whether the collection is empty or not. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AS1155)